### PR TITLE
Use FetchContent and add googletest

### PR DESCRIPTION
**Pull request by doruirimescu**
Use FetchContent to download doctest and googletest. Thus, remove submodules.
https://bewagner.net/programming/2020/05/02/cmake-fetchcontent/

Let user decide which testing framework to use

### CODE REVIEW

File: cmake/Warnings.cmake [hunk 1]
```diff
@@ -45,7 +45,7 @@ function(target_set_warnings)
       elseif(WGCC)
         list(APPEND WarningFlags "-Wall" "-Wextra" "-Wpedantic")
       elseif(WCLANG)
-        list(APPEND WarningFlags "-Wall" "-Weverything" "-Wpedantic")
+        list(APPEND WarningFlags "-Wall" "-Weverything" "-Wpedantic" "-Wno-global-constructors" "-Wno-weak-vtables")
```
Comment (doruirimescu) [inline cmake/Warnings.cmake:48]: This makes it possible to compile googletest with clang.

File: setup.sh [hunk 2]
```diff
@@ -14,9 +14,5 @@ else
 	git commit --amend --author="$name <$email>" 
 fi
 
-# Initialize submodules: This should already be done when cloning, but there are ways to muck it
```
Comment (doruirimescu) [inline setup.sh:17]: no submodules needed since we are fetching content.

**Comment by bsamseth** [comment]
Hi! Thank you for contributing this PR. FetchContent looks like a really great addition and something I definitely want to incorporate into the project.

However, I won't be adding back googletest. I used to have that as the testing framework, but I went away from it. I appreciate that there are different opinions on this, but I simply prefer doctest. While having both is an option too, you'd then have to delete lots of code whenever you make a repo from this template, which is not something I'm keen on. There are plenty of other similar templates with googletest, and it's not hard to switch from one to the other if you really want to (as demonstrated here by you). 

If you remove the code for googletest and leave just the changes to use FetchContent I will be happy to merge the PR 👍🏻

**Comment by doruirimescu** [comment]
> Hi! Thank you for contributing this PR. FetchContent looks like a really great addition and something I definitely want to incorporate into the project.
> 
> However, I won't be adding back googletest. I used to have that as the testing framework, but I went away from it. I appreciate that there are different opinions on this, but I simply prefer doctest. While having both is an option too, you'd then have to delete lots of code whenever you make a repo from this template, which is not something I'm keen on. There are plenty of other similar templates with googletest, and it's not hard to switch from one to the other if you really want to (as demonstrated here by you).
> 
> If you remove the code for googletest and leave just the changes to use FetchContent I will be happy to merge the PR 👍🏻

Hello ! 
Thanks for the review ! Ya you are right, it is quite cumbersome for a new user to make only googletest-based template. I will then make another branch, which only adds FetchContent and PR from there, so that you'd only get the relevant commits.
Best wishes,
Doru

**Pull request closed without merging**